service/controller/app/v1/resource/chart/desired_test.go

```diff
@@ -0,0 +1,226 @@
+package chart
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned/fake"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+)
+
+func TestResource_GetDesiredState(t *testing.T) {
+
+	tests := []struct {
+		name          string
+		appObj        *v1alpha1.App
```

Comment:
```suggestion
		obj        *v1alpha1.App
```

Convention is we use obj here because its the object received from the k8s informer.